algorithms/kernel/assocrules/assoc_rules_apriori_tree.i

```diff
@@ -157,7 +157,7 @@ struct hash_tree
         if (!is_child(node_idx, hash_res))
         {
             /* Update children mask of the root node if needed */
-            root[node_idx].mask |= (1 << hash_res);
+            root[node_idx].mask |= (__int64)(1) << hash_res;
```

Comment:
Too much braces. It's better to have just __int64(1). Anyway, I'm Ok with the change.

Reply:
Semantically it's different construtions. We must put type inside () due it's not 1 word.

Reply:
(Type)1  - c-style cast
Type(1)  construction of Type with passed value to ctor's parameter

Reply:
replaced by static_cast